### DISCUSSION TITLE
Stateless button, single click, direct switch activation

### DISCUSSION THREAD

**Discussion by dejan-kosak** [RavenSystem/esp-homekit-devices#1732] [Q&A] [answered]
Hi All,

I have Shelly1 set up as stateless button with three actions. 
- Single click is triggering action which is set up in “home” app. This action will eventually turn on switch in same Shelly1 device.
- Double and long click will trigger other devices. 

What I don’t like is the delay. From time when I press button and when hub turn switch on, can take 1-2 seconds.

Question: Can Single click on a button trigger Shelly1 switch directly, and only allow Double and Long click to be set up in “home” app?

**Comment by RavenSystem** [comment, marked as answer]
Single press of Stateless Button will not work. Instead, single press is used to manage switch and relay.

```json
{"a":[{"0":{"r":[{"g":4}]},"1":{"r":[{"g":4,"v":1}]},"b":[{"g":5,"p":0,"i":1}]},{"t":3,"f1":[{"g":5,"t":2}],"f2":[{"g":5,"t":3}]}]}
```

**Reply by dejan-kosak** [reply]
Wow, that was quick. Exactly what I needed. Thank you!
